TELCODOCS#2004: Day 2 Operations - Security Doc (#86603)
openshift/openshift-docs: modules/telco-security-rbac-overview.adoc

// Module included in the following assemblies:
//
// * edge_computing/day_2_core_cnf_clusters/observability/telco-security-basics.adoc

:_mod-docs-content-type: CONCEPT
[id="telco-security-rbac-overview_{context}"]
= RBAC overview

Role-based access control (RBAC) objects determine whether a user is allowed to perform a given action within a project. 

Cluster administrators can use the cluster roles and bindings to control who has various access levels to the {product-title} platform itself and all projects.

Developers can use local roles and bindings to control who has access to their projects. Note that authorization is a separate step from authentication, which is more about determining the identity of who is taking the action.

Authorization is managed using the following authorization objects:

> Rules:: Sets of permitted actions on specific objects. For example, determining whether a user or service account can `create` pods. Each rule specifies an API resource, the resource within that API, and the allowed action.

Reviewer: Do we want a full sentence here?
Probably don't need the marl-up on create in this case. 
```suggestion
Rules:: Sets of permitted actions on specific objects. For example, a rule can determine whether a user or service account can create pods. Each rule specifies an API resource, the resource within that API, and the allowed action.
```

Author: Updated as per the suggestion.

> Roles:: Collections of rules that define what actions users or groups can perform. You can associate or bind rules to multiple users or groups. A role file may contain one or more rules specifying the actions and resources allowed for that role. 

Reviewer: ```suggestion
Roles:: Roles are collections of rules that define what actions users or groups can perform. You can associate or bind rules to multiple users or groups. A role file can contain one or more rules that specify the actions and resources allowed for that role. 
```

Author: Updated as per the suggestion.

+
Roles are categorized into the following types:

* Cluster roles: You can define cluster roles at the cluster level. They are not tied to a single namespace. They can apply across all namespaces or specific namespaces when you bind them to users, groups, or service accounts.
* Project roles: You can create project roles within a specific namespace, and they only apply to that namespace. You can assign permissions to specific users to create roles and role bindings within their namespace, ensuring they do not affect other namespaces.

> Bindings:: Associations between users and/or groups with a role. You can create a role binding to connect the rules in a role to a specific user ID or group. This brings together the role and the user or group, defining what actions they can perform.

Reviewer: ```suggestion
Bindings:: Bindings are associations between users and/or groups with a role. You can create a role binding to connect the rules in a role to a specific user ID or group. This brings together the role and the user or group, defining what actions they can perform.
```

Author: Updated as per the suggestions.
